src/ucs/sys/ucs_netlink.c

```diff
@@ -0,0 +1,248 @@
+/**
+* Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2024. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "ucs_netlink.h"
+
+#include <ucs/debug/log.h>
+#include <ucs/debug/memtrack_int.h>
+#include <ucs/sys/compiler.h>
+#include <ucs/sys/sock.h>
+#include <ucs/type/status.h>
+
+#include <errno.h>
+#include <linux/netlink.h>
+#include <linux/rtnetlink.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+
+/* *******************************************************
+ * General Netlink utilities                             *
+ * ***************************************************** */
+
+static ucs_status_t ucs_netlink_socket_init(int *fd, int protocol)
+{
+    struct sockaddr_nl sa = {0};
+    ucs_status_t status;
+
+    status = ucs_socket_create(AF_NETLINK, SOCK_RAW, protocol, fd);
+    if (status != UCS_OK) {
+        ucs_error("failed to create netlink socket (%s)",
+                  ucs_status_string(status));
+        goto err;
+    }
+
+    sa.nl_family = AF_NETLINK;
+
+    if (bind(*fd, (struct sockaddr*)&sa, sizeof(sa)) < 0) {
+        ucs_error("failed to bind netlink socket %d", *fd);
+        status = UCS_ERR_IO_ERROR;
+        goto err_close_socket;
+    }
+
+    return UCS_OK;
+
+err_close_socket:
+    ucs_close_fd(fd);
+err:
+    *fd = -1;
+    return status;
+}
+
+ucs_status_t ucs_netlink_send_cmd(int protocol, unsigned short nlmsg_type,
+                                  void *nl_protocol_hdr,
+                                  size_t nl_protocol_hdr_size,
+                                  char *recv_msg_buf, size_t *recv_msg_buf_len)
+{
+    struct nlmsghdr nlh = {0};
```

Comment:
`nlh` -> more descriptive name?

Reply:
Is init really needed?

Reply:
Yes, it is, because other fields should be initialized to 0